test/e2e/crd_admission_test.go

```diff
@@ -0,0 +1,129 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/api/features"
+)
+
+// Test_GatewayAPICRDAdmissionOutsideCluster verifies that the ingress operator's ValidatingAdmissionPolicy
+// fails admission requests to modify Gateway API CRDs for a user which resides outside the cluster.
+func Test_GatewayAPICRDAdmissionOutsideCluster(t *testing.T) {
+	if gwapiEnabled, err := isFeatureGateEnabled(features.FeatureGateGatewayAPI); err != nil {
+		t.Fatalf("Failed to get Gateway API feature gate: %v", err)
+	} else if !gwapiEnabled {
+		t.Skipf("Skipping Test_GatewayAPICRDAdmissionOutsideCluster when GatewayAPI featuregate is not enabled")
```

Comment:
The VAP is being deployed in the manifests alongside the CIO, right? So I wouldn't expect the feature gate to be required here: the VAP behavior should be present whether or not someone enables the GWAPI feature, as we are not intending to allow any other entity to manage these APIs. :thinking: